tests/squirrel.test.ts

```diff
@@ -0,0 +1,88 @@
+import Squirrel from "../src/animals/Squirrel";
+
+test("should log an error if  squirrel age is negative", () => {
+    expect(() => {
+        const lion = new Squirrel("Kiki", -5, true, "Cedar", 5, 8);
+    }).toThrow("Age cannot be negative");
+});
+
+test("should log an error if name is an empty string", () => {
+    expect(() => {
+        const lion = new Squirrel("", 5, true, "Cedar", 5, 8);
+    }).toThrow("Name cannot be empty");
+});
+
+test("should log an error if treeAge is negative", () => {
+    expect(() => {
+        const squirrel = new Squirrel("Kiki", 5, true, "Cedar", -5, 56);
```

Comment:
if the logic for checking the name is in a parent class ( Aniimal for example) maybe it is redundant to test each and every animal type if this logic is correct. One animal type should be sufficient to test the method